fc_util/src/arg_parser.rs

```diff
@@ -0,0 +1,586 @@
+        }
+    }
+
+    pub fn value(&self, arg_name: &'static str) -> Option<String> {
+        self.args_matches.get(arg_name).unwrap().user_value.clone()
+    }
+
+    pub fn extra_args(self) -> Vec<String> {
+        self.extra_args
+    }
+
+    pub fn split(args: &[String]) -> (&[String], &[String]) {
+        if let Some(index) = args.iter().position(|arg| arg == DOUBLE_DASH) {
+            return (&args[0..index], &args[index + 1..args.len()]);
+        }
+
+        (&args[0..args.len()], &[])
+    }
+
+    pub fn parse(mut self) -> Result<Self> {
+        let mut args: Vec<String> = env::args().collect();
+        // Remove the first element of `args` as it should be the binary name.
+        args.remove(0);
+
+        let (args, extra_args) = ArgParser::split(&args);
+        self.extra_args = extra_args.to_vec();
+
+        self.get_matches(args)
+    }
+
+    fn check_requirements(&self, args: &[String]) -> Result<()> {
+        for arg_info in self.args_matches.iter() {
+            // The arguments that are marked `required` must be provided by user.
+            if arg_info.required && arg_info.user_value.is_none() {
+                return Err(Error::MissingArgument(arg_info.name.to_string()));
+            }
+            // For the arguments that require a specific argument to be also present in the list
+            // of arguments provided by user, search for that argument.
+            if arg_info.user_value.is_some() {
+                if let Some(arg_name) = arg_info.requires {
+                    if !args.contains(&(format!("--{}", arg_name))) {
+                        return Err(Error::MissingArgument(arg_name.to_string()));
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn is_valid(&self, arg: &str) -> Result<()> {
+        if !arg.starts_with(DOUBLE_DASH) {
+            return Err(Error::UnexpectedArgument(arg.to_string()));
+        }
+        let arg_name = &arg[DOUBLE_DASH.len()..];
+
+        // Check if the argument is an expected one.
+        if !self.args_matches.contains(arg_name) {
+            return Err(Error::UnexpectedArgument(arg_name.to_string()));
+        }
+
+        // Check if the argument was not provided more than once.
+        if self
+            .args_matches
+            .get(arg_name)
+            .unwrap()
+            .user_value
+            .is_some()
+        {
+            return Err(Error::DuplicatedArgument(arg_name.to_string()));
+        }
+        Ok(())
+    }
+
+    // Populates the fields of ArgParser.
+    pub fn get_matches(mut self, args: &[String]) -> Result<Self> {
+        if args.contains(&HELP_ARG.to_string()) {
+            self.print_help();
+            process::exit(0)
+        }
+
+        let mut index = 0;
+        while index < args.len() {
+            let arg = &args[index];
+
+            self.is_valid(arg)?;
+
+            let arg_info = self.args_matches.get(&arg[DOUBLE_DASH.len()..]).unwrap();
```

Comment:
nit: I know it's safe to unwrap, but I would still prefer the `ok_or()` approach.